runtime/lua/vim/lsp/util.lua

```diff
@@ -31,6 +31,36 @@ local function split_lines(value)
   return split(value, '\n', true)
 end
 
+function M.add_hook_after(func, new_fn)
+  if func then
+    return function(...)
+      -- TODO which result?
+      func(...)
+      return new_fn(...)
+    end
+  else
+    return new_fn
+  end
+end
+
+function M.wrap_on_init(config)
+  return M.add_hook_after(config.on_init,
+    function(client, _result)
+      function client.workspace_did_change_configuration(settings)
```

Comment:
I'm not sure how useful it would be to expose this, given it's just validating the settings sent to 
```
client.notify('workspace/didChangeConfiguration', {
          settings = settings;
        })
```